models/articlesModel.js

```diff
@@ -1,45 +1,71 @@
 const db = require("../db/connection.js");
 
-const fetchArticleById = (articleId) => {
-    if (isNaN(parseInt(articleId))) {
-        return Promise.reject({
-            status: 400,
-            msg: "Bad request"
-        })
-    }
-    return db.query("SELECT * FROM ARTICLES WHERE article_id = $1", [articleId])
-    .then(({rows}) => {
-        const article = rows[0]
-        if (!article) {
-            return Promise.reject({
-                status: 404,
-                msg: `No article found for article_id ${articleId}`
-            })
-        }
-        return article
-    })
-}
+const fetchArticleById = (articleId, commentCount) => {
+	if (isNaN(parseInt(articleId))) {
```

Comment:
We should be letting SQL tell us whether this ID is valid or not. If its not a number, SQL with reject with a '22P02' code and we can handle that with our error handlers and catch blocks. This is essentially repeated code because we already have the functionality given to us by SQL